utils/perl5lib/Module/ModuleLoader.pm

```diff
@@ -655,15 +655,26 @@ sub loadSoftModules()
 
     my %newbuildenv;
 
+	# ndk: add a loop to see if our above module adjusting _removed_
```

Comment:
After module commands (removing/loading), the script looks at the env variables.  Before and After.
Any new ones, it adds to the list.  Any changes to existing vars, it changes.
Any env vars that are _removed_ it would not previously remove them, which caused several issues.
This fix removes them.  Surely, new cime does not do things in this way.